### PR TITLE
whitelist integration

### PR DESCRIPTION
**Pull request by davidangb**
download a whitelist from configurable GCS location; validate that current user is in the whitelist.

### CODE REVIEW

File: function/auth.js
Comment (gpolumbo-broad) [inline function/auth.js:31]: Add contact info or add a `TODO`

File: function/config/default.json
Comment (gpolumbo-broad) [inline function/config/default.json:4]: Will testers/developers need to do anything in this bucket?  Are permissions set appropriately on this bucket to allow (or disallow) that interaction?

**Review by gpolumbo-broad — CHANGES_REQUESTED** [review]

File: function/auth.js
Comment (davidangb) [reply function/auth.js:31]: the TODO is at line 29 :)

File: function/config/default.json
Comment (davidangb) [reply function/config/default.json:4]: testers/developers don't need to do anything now, and the bucket is properly permissioned.  We are waiting on SIA approval and the final hosting projects to be created; once that's in place and deploy scripts are running,  I'll  go  back and update  all of this config.

**Review by gpolumbo-broad — APPROVED** [review]

**Comment by dvoet** [comment]
alternatively, you could use a group in sam for you whitelist

**Pull request merged**
